spec/requests/cinema_halls_request_rspec.rb

```diff
@@ -4,41 +4,83 @@
 
 RSpec.describe 'CinemaHalls', type: :request do
   describe 'GET /cinema_halls' do
+    let!(:cinema_hall) { CinemaHall.create(name: 'Lalka', number_of_seats: 100) }
+
     it 'returns http success' do
       get '/cinema_halls'
       expect(response).to have_http_status(200)
     end
   end
-end
 
-RSpec.describe 'CinemaHalls', type: :request do
   describe 'POST /cinema_halls' do
-    scenario 'valid cinema_hall attribute' do
-      post '/cinema_halls', params: {
-        cinema_hall: {
-          number_of_seats: 50,
-          name: 'Calineczka'
-        }
-      }
+    context 'valid cinema_hall attributes' do
+      let(:cinema_hall) { CinemaHall.new(name: 'Lalka', number_of_seats: 100) }
+
+      it 'valid http status' do
+        post '/cinema_halls', params: { name: cinema_hall.name, number_of_seats: cinema_hall.number_of_seats }
```

Comment:
You have repetitive `post`. Put it into `subject` variable to just use it into `it` block like:
```ruby
  context 'valid cinema_hall attributes' do
    subject { '/cinema_halls', params: { name: cinema_hall.name, number_of_seats: cinema_hall.number_of_seats } }
    
    it 'valid http status' do
      subject
      expect(response).to have_http_status(201)
    end
  end
```

I would extract `params` to `let` like you did with `cinema_hall` :)

Reply:
I added subject and factory_bot :)